addons/account_edi_ubl/models/account_edi_format_en_16931.py

```diff
@@ -0,0 +1,175 @@
+        ''' Checks if the xml can be imported with this EN16931 implementation.
+            Note that you need to override _create_invoice_from_xml_tree/_update_invoice_from_xml_tree or _import_ubl
+            if you want to customize import, otherwise generic import will be used.
+
+            TO OVERRIDE
+        '''
+        return False  # we must check self.code which does not exist here because we're in an abstract format.
+
+    def _get_en_16931_values(self, invoice):
+        self.ensure_one()
+
+        def convert_monetary(amount, from_currency):
+            # All monetary should be in the invoice currency, except for vat total
+            return from_currency._convert(amount, invoice.currency_id, invoice.company_id, invoice.invoice_date)
+
+        def get_tax_total():
+            breakdown = {}
+            for line in invoice.invoice_line_ids.filtered(lambda line: not line.display_type):
+                if line.tax_ids:
+                    price_unit_wo_discount = line.price_unit * (1 - (line.discount / 100.0))
+                    line_taxes = line.tax_ids.compute_all(
+                        price_unit_wo_discount,
+                        quantity=line.quantity,
+                        product=line.product_id,
+                        partner=invoice.partner_id,
+                        currency=line.currency_id,
+                        is_refund=invoice.move_type in ('out_refund', 'in_refund'))['taxes']
+                    for tax in line_taxes:
+                        tax_category = 'S' if tax['amount'] else 'Z'
+                        tax_percent = self.env['account.tax'].browse(tax['id']).amount
+                        breakdown.setdefault((tax_category, tax_percent), {'base': 0, 'amount': 0})
+                        breakdown[(tax_category, tax_percent)]['base'] += tax['base']
+                        breakdown[(tax_category, tax_percent)]['amount'] += tax['amount']
+                else:
+                    breakdown.setdefault(('Z', 0), {'base': 0, 'amount': 0})
+                    breakdown[('Z', 0)]['base'] += line.price_subtotal
+
+            sign = -1 if invoice.move_type in ('out_refund', 'in_refund') else 1
+            return {'amount': invoice.amount_tax,
+                    'seller_currency': invoice.company_id.currency_id,
+                    'amount_seller_currency': invoice.currency_id._convert(invoice.amount_tax_signed * sign, invoice.company_id.currency_id, invoice.company_id, invoice.date),
+                    'breakdown': breakdown}
+
+        values = self._get_ubl_values(invoice)
+        values.update({
+            'customization_id': 'urn:cen.eu:en16931:2017#compliant#urn:fdc:peppol.eu:2017:poacc:billing:3.0',
+            'profile_id': 'urn:fdc:peppol.eu:2017:poacc:billing:01:1.0',
+            'due_date': fields.Date.to_string(invoice.invoice_date_due),
+            'invoice_date': fields.Date.to_string(invoice.invoice_date),
+            'tax_total': get_tax_total(),
```

Comment:
I'm not sure I really understand what you want.
_compute_invoice_taxes_by_group and get_tax_total have different result.

The way taxes are formatted is only required by EN16931, hence this method in the parent class so that all the format deriving from EN16931 will use this instead of re-defining each time.